smash/tests/core/simulation/test_run.py

```diff
@@ -96,3 +96,49 @@ def test_multiple_forward_run():
 
     # % Check that forward run discharge is equivalent to multiple forward run discharge
     assert np.allclose(frq, mfr.q, atol=1e-06, equal_nan=True), "multiple_forward_run.q"
+
+
+def test_forward_run_mlp():
+    ncpu = min(5, max(1, os.cpu_count() - 1))
+
+    # % Test some multi layer perceptron models versus classical models
+    # % TODO: add ode and ode_mlp once implicit Euler for neural ode is implemented
+    mlp_to_cls_structure = {
+        "zero-gr4_mlp-lr": "zero-gr4-lr",
+        "zero-gr5_mlp-lr": "zero-gr5-lr",
+        "zero-gr6_mlp-lr": "zero-gr6-lr",
+        "zero-grd_mlp-lr": "zero-grd-lr",
+        "zero-grc_mlp-lr": "zero-grc-lr",
+        "zero-loieau_mlp-lr": "zero-loieau-lr",
+    }
+    all_structure = [model.setup.structure for model in pytest.model_structure]
+    for mlp_structure, cls_structure in mlp_to_cls_structure.items():
+        mlp_idx = all_structure.index(mlp_structure)
+        cls_idx = all_structure.index(cls_structure)
+        mlp_model = pytest.model_structure[mlp_idx]
+        cls_model = pytest.model_structure[cls_idx]
+
+        # % Reset NN param values to zeros
+        mlp_model.set_nn_parameters_weight(initializer="zeros")
+        mlp_model.set_nn_parameters_bias(initializer="zeros")
```

Comment:
J'ai vu le précécent PR par rapport à l'initilisation des poids et bias ? Une explication pourquoi la première méthode n'a pas marché ? Un problème de copy ?

Reply:
Oui, c'était parce que les params de NN ont été initialisés avant ces tests (dans forward_run et dans optimize):
```python
if model.setup.n_layers > 0:
            model.set_nn_parameters_weight(initializer="glorot_normal", random_state=11)
```
Donc la liste pytest.model_structure a été modifié avec des params de NN non-nuls avant ce test. Ces lignes sont donc pour remettre ces valeurs à 0, donc égale au modèle original.

Reply:
La première méthode n'a pas marché car quand tu devrais faire les optims in-place sur un copie de model dans la boucle de mappings. On peut faire un optim non-inplace (smash.optimize) mais du coup ça crée encore un autre copie, donc je préfère plutôt la méthode actuelle.

Reply:
Oui exactement.